src/sentry/incidents/logic.py

```diff
@@ -2,14 +2,16 @@
 
```

Comment:
this file is in the ignore list so none of this work is checked -- fix that first and *then* improve the type annotations

Reply:
This started out smaller in scope. I was checking the work as I went with local `mypy` runs and PyCharm warnings, not necessarily intending to get as far as removing the file from the ignore list. When I did try that (earlier in this branch's history), there were about ~100 more failures, which seemed like too much to tackle at the time. I'd love to devote more time later to getting `logic.py` fully checked, but I was hoping this branch would be mergeable in its current state as intermediate progress.

Would you agree with that? Or are you saying that we shouldn't be adding type annotations without CI coverage? They may contain some mistakes, but from the offline checks, I'd say I have a high degree of confidence that these will help clarity more than harm it in the short term. But I'll understand if you don't think it's worth the risk.

Reply:
without the progress being checked I have no confidence in the improvements here (and no confidence that they won't regress).  we have a mechanism to make incremental improvements -- just need to use it